### ISSUE TITLE
nginx parser does not capture multiple entries in http_x_forwarded_for-header when it contains white-spaces

### ISSUE THREAD

**Issue by mkoertgen** [fluent/fluentd#3420]
Check [CONTRIBUTING guideline](https://github.com/fluent/fluentd/blob/master/CONTRIBUTING.md) first and here is the list to help us investigate the problem.

**Describe the bug**

The [X-Forwarded-For](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For) allows a list of forwards which may include whitespace, e.g. `client, proxy1, proxy2`
The [current regexp in the nginx-parser](https://docs.fluentd.org/parser/nginx#regexp-patterns) does not capture that.

**To Reproduce**

Have an nginx log with multiple proxies and the log will not be captured.

Check with fluentular:

- [fluentular - current nginx-regexp](http://fluentular.herokuapp.com/parse?regexp=%5E%28%3F%3Cremote%3E%5B%5E+%5D*%29+%28%3F%3Chost%3E%5B%5E+%5D*%29+%28%3F%3Cuser%3E%5B%5E+%5D*%29+%5C%5B%28%3F%3Ctime%3E%5B%5E%5C%5D%5D*%29%5C%5D+%22%28%3F%3Cmethod%3E%5CS%2B%29%28%3F%3A+%2B%28%3F%3Cpath%3E%5B%5E%5C%22%5D*%3F%29%28%3F%3A+%2B%5CS*%29%3F%29%3F%22+%28%3F%3Ccode%3E%5B%5E+%5D*%29+%28%3F%3Csize%3E%5B%5E+%5D*%29%28%3F%3A+%22%28%3F%3Creferer%3E%5B%5E%5C%22%5D*%29%22+%22%28%3F%3Cagent%3E%5B%5E%5C%22%5D*%29%22%28%3F%3A%5Cs%2B%28%3F%3Chttp_x_forwarded_for%3E%5B%5E+%5D%2B%29%29%3F%29%3F%24&input=clientip+-+-+%5B10%2FJun%2F2021%3A07%3A01%3A30+%2B0000%5D+%22GET+%2Fsome%2Fpath+HTTP%2F1.1%22+200+26664+%22https%3A%2F%2Fsome-refererrer%22+%22user-agent%22+%22client1%2Cproxy1%2Cproxy2%22&time_format=)
- [fluentular - suggestion to fix](http://fluentular.herokuapp.com/parse?regexp=%5E%28%3F%3Cremote%3E%5B%5E+%5D*%29+%28%3F%3Chost%3E%5B%5E+%5D*%29+%28%3F%3Cuser%3E%5B%5E+%5D*%29+%5C%5B%28%3F%3Ctime%3E%5B%5E%5C%5D%5D*%29%5C%5D+%22%28%3F%3Cmethod%3E%5CS%2B%29%28%3F%3A+%2B%28%3F%3Cpath%3E%5B%5E%5C%22%5D*%3F%29%28%3F%3A+%2B%5CS*%29%3F%29%3F%22+%28%3F%3Ccode%3E%5B%5E+%5D*%29+%28%3F%3Csize%3E%5B%5E+%5D*%29%28%3F%3A+%22%28%3F%3Creferer%3E%5B%5E%5C%22%5D*%29%22+%22%28%3F%3Cagent%3E%5B%5E%5C%22%5D*%29%22%28%3F%3A%5Cs%2B%22%28%3F%3Chttp_x_forwarded_for%3E%5B%5E%5C%22%5D%2B%29%22%29%3F%29%3F%24&input=clientip+-+-+%5B10%2FJun%2F2021%3A07%3A01%3A30+%2B0000%5D+%22GET+%2Fsome%2Fpath+HTTP%2F1.1%22+200+26664+%22https%3A%2F%2Fsome-refererrer%22+%22user-agent%22+%22client1%2C+proxy1%2C+proxy2%22&time_format=)

**Expected behavior**
The above log line can be captured

**Your Environment**

docker image: fluent/fluentd-kubernetes-daemonset:v1.12.3-debian-elasticsearch7-1.0

**Comment by mkoertgen** [comment]
A slight modification of the regexp would help:

```regexp
^(?<remote>[^ ]*) (?<host>[^ ]*) (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^\"]*?)(?: +\S*)?)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>[^\"]*)"(?:\s+"(?<http_x_forwarded_for>[^\"]+)")?)?$
```

This captures an optional forward-for-header enclosed by quotes.

**Comment by mkoertgen** [comment]
Seems to be addressed in 1.13.0 with https://github.com/fluent/fluentd/pull/2171

**Comment by mkoertgen** [comment]
Waiting for docker image for 1.13.0 to become available, cf.:
- https://hub.docker.com/r/fluent/fluentd/tags
